### ISSUE TITLE
Search non functional on Android tv

### ISSUE THREAD

**Issue by ltomes** [TeamNewPipe/NewPipe#3751]
### Version

-
Latest on fdroid 0.19.4
### Expected behavior

Searches are functional on Android tv
### Actual behaviour

Search is not functional with newpipe on Android tv devices. Searches result in an error toast message.
### Logs

Other info:
Device: Nvidia shield

**Comment by Stypox** [comment]
There is not enough information. Please provide the logs from the toast message.

**Comment by test2a** [comment]
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ GB
* __Service:__ none
* __Version:__ 0.19.5
* __OS:__ Linux Android 7.0 - 24


<details><summary><b>Crash log</b></summary><p>

```
java.lang.ArrayIndexOutOfBoundsException: length=25; index=-1
	at java.util.ArrayList.get(ArrayList.java:413)
	at org.schabi.newpipe.fragments.list.search.SuggestionListAdapter.getItem(SuggestionListAdapter.java:82)
	at org.schabi.newpipe.fragments.list.search.SearchFragment.getSuggestionMovementFlags(SearchFragment.java:1023)
	at org.schabi.newpipe.fragments.list.search.SearchFragment$1.getMovementFlags(SearchFragment.java:329)
	at androidx.recyclerview.widget.ItemTouchHelper$Callback.getAbsoluteMovementFlags(ItemTouchHelper.java:1593)
	at androidx.recyclerview.widget.ItemTouchHelper.checkSelectForSwipe(ItemTouchHelper.java:991)
	at androidx.recyclerview.widget.ItemTouchHelper$2.onInterceptTouchEvent(ItemTouchHelper.java:351)
	at androidx.recyclerview.widget.RecyclerView.findInterceptingOnItemTouchListener(RecyclerView.java:3151)
	at androidx.recyclerview.widget.RecyclerView.onInterceptTouchEvent(RecyclerView.java:3170)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2212)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2669)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2358)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:611)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1817)
	at android.app.Activity.dispatchTouchEvent(Activity.java:3102)
	at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
	at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:433)
	at android.view.View.dispatchPointerEvent(View.java:10168)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4440)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4308)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3855)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3908)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3874)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4001)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3882)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4058)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3855)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3908)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3874)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3882)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3855)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6216)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6190)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6151)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6319)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
	at android.view.InputEventReceiver.nativeConsumeBatchedInputEvents(Native Method)
	at android.view.InputEventReceiver.consumeBatchedInputEvents(InputEventReceiver.java:176)
	at android.view.ViewRootImpl.doConsumeBatchedInput(ViewRootImpl.java:6290)
	at android.view.ViewRootImpl$ConsumeBatchedInputRunnable.run(ViewRootImpl.java:6342)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:871)
	at android.view.Choreographer.doCallbacks(Choreographer.java:683)
	at android.view.Choreographer.doFrame(Choreographer.java:613)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:857)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6153)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:868)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:758)

```
</p></details>
<hr>

**Comment by test2a** [comment]
i have attached my crash log. my search is not working for the past 2-3 days. attached here is screenrecord of search not working on my phone
![000s1591539282-videorecord-2020-6-7-19-44-42](https://user-images.githubusercontent.com/39449563/83971211-c8f35480-a8f7-11ea-8588-f7a33018c5ab.gif)

**Comment by Stypox** [comment]
@ltomes @test2a could you test the apk provided in #3759 and see if it fixes the issue?

**Comment by test2a** [comment]
@stypox installed the apk but exact same search issue. Unable to type or properly search. One thing, in the original app, dismissing search history would show crash report but not in test apk.

Where are the debug logs stored?

**Comment by Rub-z** [comment]
Hi, I was having the exact issue but it was because I was using NewPipe on a network that had a PiHole (DNS Filtering). Getting the admin to white list 'suggestqueries.google.com' fixed the issue. 

I thought I'd add here in case it helps anyone else.

**Comment by B0pol** [comment]
Actually you can block suggestqueries.google.com, just go to settings and disable suggestions. But we should detect if it's blocked to prevent that

**Comment by wb9688** [comment]
Imho we just shouldn't catch the `IOException` like that for search suggestions.

**Comment by ltomes** [comment]
I can't test this at this time, my android tv is packed up. I can test this sometime in the future though if the issue is resolved.

**Comment by Stypox** [comment]
> I can't test this at this time, my android tv is packed up. I can test this sometime in the future though if the issue is resolved.

Ok, don't worry. Let's assume this was fixed by #6558. If it wasn't and this is still reproducible, feel free to reopen (you or anyone else).
